File: Medium/3-longest-substring-without-repeating-characters/try-6.py
/*
 * Problem #3: Longest Substring Without Repeating Characters
 * Difficulty: Medium
 * Submission: Try 6
 * status: Accepted
 * Language: python3
 * Date: 07/12/2024, 22:20:02
 * Link: https://leetcode.com/problems/longest-substring-without-repeating-characters/
 */

class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        map=Counter()
        end=0
        intial=0
        maximum=0
        while(end<len(s)):
            map[s[end]]+=1
            while(map[s[end]]>1):
                map[s[intial]]-=1
                intial+=1
            maximum=max(maximum,end-intial+1)
            end+=1
        return maximum
        

File: Medium/53-maximum-subarray/try-6.cpp
/*
 * Problem #53: Maximum Subarray
 * Difficulty: Medium
 * Submission: Try 6
 * status: Accepted
 * Language: cpp
 * Date: 27/01/2025, 19:16:21
 * Link: https://leetcode.com/problems/maximum-subarray/
 */

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = 0,re = INT_MIN;
        for(int i = 0; i<nums.size();i++){
            sum = sum + nums[i];
            re = max(re,sum);
            if(sum < 0){
                sum = 0;
            }
        }
        return re;
    }
};
